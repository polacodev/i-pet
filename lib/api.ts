import { decode } from 'base64-arraybuffer';

import { supabase } from '@/lib/supabase';
import { PetProps } from '@/types/pet';
import { Session } from '@/types/session';
import { UserProfileProps } from '@/types/user';

export const getPetList = async (userId: string | null) => {
  let petList: PetProps[] = [];
  const { data, error } = await supabase
    .from('pets')
    .select(
      `
      id,
      inserted_at,
      pet_image,
      pet_age,
      pet_breed,
      pet_gender,
      pet_medical_condition,
      pet_name,
      pet_type,
      user_id,
      profiles (
        id,
        email,
        country_code,
        phone
      )
      `,
    )
    .eq('user_id', userId);

  // if (error) return { data, error };
  // if (data) {
  petList = data;
  return { data: petList, error };
  // }
  // return { data, error };
};

export const getPetById = async (petId: string) => {
  const { data, error } = await supabase
    .from<any, 'public'>('pets')
    .select(
      `
      id,
      inserted_at,
      pet_image,
      pet_age,
      pet_breed,
      pet_gender,
      pet_medical_condition,
      pet_name,
      pet_type,
      user_id,
      profiles (
        id,
        email,
        country_code,
        phone
      )
    `,
    )
    .eq('id', petId)
    .single();
  return { data, error };
};

export const deletePetById = async (petId: string) => {
  const { error } = await supabase.from('pets').delete().eq('id', petId);
  return { error };
};

export const insertPet = async (newPet: any) => {
  const { error } = await supabase.from('pets').upsert(newPet);
  return { error };
};

export const storePetImage = async (filePath: string, code: any, contentType: string) => {
  const { data, error } = await supabase.storage
    .from('pets')
    .upload(filePath, decode(code), { contentType });
  return { data, error };
};

export const deletePetImage = async (petImage: string) => {
  const { data, error } = await supabase.storage.from('pets').remove([petImage]);
  return { data, error };
};

export const signInWithPassword = async (email: string, password: string) => {
  const { data, error } = await supabase.auth.signInWithPassword({
    email,
    password,
  });
  return { data, error };
};

export const signUpWithEmailPassword = async (email: string, password: string) => {
  const { data, error } = await supabase.auth.signUp({ email, password });
  return { data, error };
};

export const saveUserProfile = async (user: UserProfileProps) => {
  const { data, error } = await supabase.from('profiles').upsert(user);
  return { data, error };
};

export const signOut = async () => {
  await supabase.auth.signOut();
};

export const getSession = async () => {
  return await supabase.auth.getSession();
};

export const onAuthStateChange = (setSession: (session: Session | null) => void) => {
  return supabase.auth.onAuthStateChange((_event, session) => {
    setSession(session);
  });
};
