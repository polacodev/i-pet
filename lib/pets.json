[
  {
    "id": "550e8400-e29b-41d4-a716-446655440000",
    "petName": "Bella",
    "petType": "Dog",
    "petAge": 3,
    "petBreed": "Golden Retriever",
    "petGender": "Female",
    "petMedicalCondition": "Hip Dysplasia",
    "ownerName": "John Smith",
    "ownerPhone": "60717546"
  },
  {
    "id": "550e8400-e29b-41d4-a716-446655440001",
    "petName": "Max",
    "petType": "Dog",
    "petAge": 2,
    "petBreed": "Bulldog",
    "petGender": "Male",
    "petMedicalCondition": "Allergies",
    "ownerName": "Emily Johnson",
    "ownerPhone": "60717546"
  },
  {
    "id": "550e8400-e29b-41d4-a716-446655440002",
    "petName": "Luna",
    "petType": "Cat",
    "petAge": 1,
    "petBreed": "Siamese",
    "petGender": "Female",
    "petMedicalCondition": "Asthma",
    "ownerName": "Michael Brown",
    "ownerPhone": "60717546"
  },
  {
    "id": "550e8400-e29b-41d4-a716-446655440003",
    "petName": "Charlie",
    "petType": "Dog",
    "petAge": 5,
    "petBreed": "Labrador",
    "petGender": "Male",
    "petMedicalCondition": "Arthritis",
    "ownerName": "Jessica Davis",
    "ownerPhone": "60717546"
  },
  {
    "id": "550e8400-e29b-41d4-a716-446655440004",
    "petName": "Daisy",
    "petType": "Dog",
    "petAge": 4,
    "petBreed": "Beagle",
    "petGender": "Female",
    "petMedicalCondition": "Diabetes",
    "ownerName": "David Wilson",
    "ownerPhone": "60717546"
  }
]