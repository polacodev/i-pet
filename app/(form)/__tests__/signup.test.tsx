import { fireEvent, render, waitFor } from '@testing-library/react-native';
import '@testing-library/react-native/extend-expect';
import { router } from 'expo-router';
import React from 'react';
import { Alert } from 'react-native';

import Signup from '@/app/(form)/signup';

jest.mock('expo-router', () => ({
  router: {
    navigate: jest.fn(),
  },
}));

jest.mock('@/localizations/localization', () => ({
  localization: {
    t: jest.fn().mockImplementation((key) => key),
  },
}));

jest.mock('@/lib/api', () => ({
  signUpWithEmailPassword: jest.fn(),
  saveUserProfile: jest.fn(),
}));

describe('<Signup />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.useFakeTimers();
    // useStateMock.mockImplementation((init: boolean) => [init, setState]);
    (router.navigate as jest.Mock).mockClear();
  });

  afterEach(() => {
    jest.runOnlyPendingTimers();
    jest.useRealTimers();
    jest.resetAllMocks();
  });

  test('Signup renders correctly and Snapshot', () => {
    const { toJSON } = render(<Signup />);

    expect(toJSON()).toMatchSnapshot();
  });

  test('goToLogIn is called when login label is pressed', () => {
    const { getByTestId } = render(<Signup />);

    fireEvent.press(getByTestId('redirect-to-login-label'), 'onClick');

    expect(router.navigate).toHaveBeenCalledWith('/login');
  });

  test('Email, Password, Phone: Change event are called correctly with valid Data to call signUpWithEmailPassword and redirect to Home', async () => {
    require('@/lib/api').signUpWithEmailPassword.mockReturnValue({
      data: {
        session: {
          user: {
            id: 'testId',
            email: 'test@test.com',
            country_code: '+1',
            phone: '123456',
          },
        },
      },
      error: null,
    });

    const { getByTestId, queryByTestId } = render(<Signup />);

    // Valid Data
    const email = 'test@test.com';
    const password = 'password';
    const phone = '12345678';

    fireEvent.changeText(getByTestId('signup-email-input'), email);
    fireEvent.changeText(getByTestId('signup-password-input'), password);
    fireEvent.changeText(getByTestId('signup-phone-input'), phone);

    await waitFor(() => {
      fireEvent.press(getByTestId('pet-button'), 'onClick');
    });

    // does not render the errors because the form is submitted correctly
    expect(queryByTestId('signup-email-input-error')).toBeFalsy();
    expect(queryByTestId('signup-password-input-error')).toBeFalsy();
    expect(queryByTestId('signup-phone-input-error')).toBeFalsy();

    // got the session instance to redirect Home after a successfully signup
    expect(router.navigate).toHaveBeenCalledWith('/');
  });

  test('Email, Password, Phone: Change event are called correctly with invalid Data, the form fails because the validations and Snapshot', async () => {
    require('@/lib/api').signUpWithEmailPassword.mockReturnValue({
      data: {
        session: {
          user: {
            id: 'testId',
            email: 'test@test.com',
            country_code: '+1',
            phone: '123456',
          },
        },
      },
      error: null,
    });

    const { getByTestId, queryByTestId, toJSON } = render(<Signup />);

    // Invalid Data to make the form fail
    const email = 'test';
    const password = 'pass';
    const phone = 'abc';

    fireEvent.changeText(getByTestId('signup-email-input'), email);
    fireEvent.changeText(getByTestId('signup-password-input'), password);
    fireEvent.changeText(getByTestId('signup-phone-input'), phone);

    await waitFor(() => {
      fireEvent.press(getByTestId('pet-button'), 'onClick');
    });

    expect(toJSON()).toMatchSnapshot();

    // Error section render the errors because the form is not submitted
    expect(queryByTestId('signup-email-input-error')).toBeTruthy();
    expect(queryByTestId('signup-password-input-error')).toBeTruthy();
    expect(queryByTestId('signup-phone-input-error')).toBeTruthy();

    // got the session instance but does not navigate because the validations
    expect(router.navigate).not.toHaveBeenCalled();
  });

  test('signUpWithEmailPassword is called but the session does not contain Data, is null. Alert is displayed and does not navigate', async () => {
    require('@/lib/api').signUpWithEmailPassword.mockReturnValue({
      data: {
        session: null,
      },
      error: new Error('signUpWithEmailPassword error'),
    });

    const AlertErrorSpy = jest.spyOn(Alert, 'alert');

    const { getByTestId } = render(<Signup />);

    // Valid Data
    const email = 'test@test.com';
    const password = 'password';
    const phone = '12345678';

    fireEvent.changeText(getByTestId('signup-email-input'), email);
    fireEvent.changeText(getByTestId('signup-password-input'), password);
    fireEvent.changeText(getByTestId('signup-phone-input'), phone);

    await waitFor(() => {
      fireEvent.press(getByTestId('pet-button'), 'onClick');
    });

    expect(AlertErrorSpy).toHaveBeenCalled();
    expect(router.navigate).not.toHaveBeenCalled();
  });

  test('saveUserProfile is user profile is created successfully', async () => {
    require('@/lib/api').signUpWithEmailPassword.mockReturnValue({
      data: {
        session: {
          user: {
            id: 'testId',
            email: 'test@test.com',
            country_code: '+1',
            phone: '123456',
          },
        },
      },
      error: null,
    });

    require('@/lib/api').saveUserProfile.mockReturnValue({
      error: null,
    });

    const AlertErrorSpy = jest.spyOn(Alert, 'alert');

    const { getByTestId } = render(<Signup />);

    // Valid Data
    const email = 'test@test.com';
    const password = 'password';
    const phone = '12345678';

    fireEvent.changeText(getByTestId('signup-email-input'), email);
    fireEvent.changeText(getByTestId('signup-password-input'), password);
    fireEvent.changeText(getByTestId('signup-phone-input'), phone);

    await waitFor(() => {
      fireEvent.press(getByTestId('pet-button'), 'onClick');
    });

    expect(AlertErrorSpy).not.toHaveBeenCalled();
  });

  test('saveUserProfile request fails and return an Error', async () => {
    require('@/lib/api').signUpWithEmailPassword.mockReturnValue({
      data: {
        session: {
          user: {
            id: 'testId',
            email: 'test@test.com',
            country_code: '+1',
            phone: '123456',
          },
        },
      },
      error: null,
    });

    // saveUserProfile request fails and return an Error
    require('@/lib/api').saveUserProfile.mockReturnValue({
      error: new Error('saveUserProfile error'),
    });

    // AlertErrorSpy to get the error when saveUserProfile fails
    const AlertErrorSpy = jest.spyOn(Alert, 'alert');

    const { getByTestId } = render(<Signup />);

    // Valid Data
    const email = 'test@test.com';
    const password = 'password';
    const phone = '12345678';

    fireEvent.changeText(getByTestId('signup-email-input'), email);
    fireEvent.changeText(getByTestId('signup-password-input'), password);
    fireEvent.changeText(getByTestId('signup-phone-input'), phone);

    await waitFor(() => {
      fireEvent.press(getByTestId('pet-button'), 'onClick');
    });

    expect(AlertErrorSpy).toHaveBeenCalledTimes(1);
  });

  test('saveUserProfile request fails because is rejected the request and return a Catch Error and error is Instance of Error)', async () => {
    require('@/lib/api').signUpWithEmailPassword.mockReturnValue({
      data: {
        session: {
          user: {
            id: 'testId',
            email: 'test@test.com',
            country_code: '+1',
            phone: '123456',
          },
        },
      },
      error: null,
    });

    // saveUserProfile request fails and return an Error and is an Instance of Error
    require('@/lib/api').saveUserProfile.mockRejectedValue(
      new Error('saveUserProfile rejected error'),
    );

    // AlertErrorSpy to get the error when saveUserProfile fails
    const AlertErrorSpy = jest.spyOn(Alert, 'alert');

    const { getByTestId } = render(<Signup />);

    // Valid Data
    const email = 'test@test.com';
    const password = 'password';
    const phone = '12345678';

    fireEvent.changeText(getByTestId('signup-email-input'), email);
    fireEvent.changeText(getByTestId('signup-password-input'), password);
    fireEvent.changeText(getByTestId('signup-phone-input'), phone);

    await waitFor(() => {
      fireEvent.press(getByTestId('pet-button'), 'onClick');
    });

    // Catch Alert is active to show the error
    expect(AlertErrorSpy).toHaveBeenCalledTimes(1);
  });

  test('saveUserProfile request fails because is rejected the request and return a Catch Error and error is not Instance of Error)', async () => {
    require('@/lib/api').signUpWithEmailPassword.mockReturnValue({
      data: {
        session: {
          user: {
            id: 'testId',
            email: 'test@test.com',
            country_code: '+1',
            phone: '123456',
          },
        },
      },
      error: null,
    });

    // saveUserProfile request fails and return an Error and is not an Instance of Error
    require('@/lib/api').saveUserProfile.mockRejectedValue('not and instance of error');

    // AlertErrorSpy to get the error when saveUserProfile fails
    const AlertErrorSpy = jest.spyOn(Alert, 'alert');

    const { getByTestId } = render(<Signup />);

    // Valid Data
    const email = 'test@test.com';
    const password = 'password';
    const phone = '12345678';

    fireEvent.changeText(getByTestId('signup-email-input'), email);
    fireEvent.changeText(getByTestId('signup-password-input'), password);
    fireEvent.changeText(getByTestId('signup-phone-input'), phone);

    await waitFor(() => {
      fireEvent.press(getByTestId('pet-button'), 'onClick');
    });

    // Catch Alert is not active because the Error is not an Instance of Error
    expect(AlertErrorSpy).not.toHaveBeenCalledTimes(1);
  });
});
