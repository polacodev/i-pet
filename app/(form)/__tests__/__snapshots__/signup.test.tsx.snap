// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> Email, Password, Phone: Change event are called correctly with invalid Data, the form fails because the validations and Snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#f6f6f6",
      },
      {
        "width": "100%",
      },
    ]
  }
  testID="pet-view"
>
  <View
    style={
      [
        {
          "backgroundColor": "#f6f6f6",
        },
        {
          "alignItems": "center",
          "gap": 5,
          "justifyContent": "center",
        },
      ]
    }
    testID="pet-view"
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          {
            "paddingHorizontal": 50,
            "paddingVertical": 20,
          },
        ]
      }
      testID="pet-text"
    />
    <TextInput
      autoCapitalize="none"
      defaultValue=""
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholderTextColor="#8e8e8e"
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "borderColor": "#22D3EE",
            "borderRadius": 10,
            "borderWidth": 1,
            "height": 45,
            "paddingLeft": 25,
            "paddingRight": 25,
            "width": "85%",
          },
          undefined,
        ]
      }
      testID="signup-email-input"
      value="test"
    />
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 14,
            "lineHeight": 20,
          },
          {
            "color": "red",
            "fontSize": 11,
            "margin": -5,
          },
        ]
      }
      testID="signup-email-input-error"
    >
      value does not match email format
    </Text>
    <TextInput
      autoCapitalize="none"
      defaultValue=""
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholderTextColor="#8e8e8e"
      secureTextEntry={true}
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "borderColor": "#22D3EE",
            "borderRadius": 10,
            "borderWidth": 1,
            "height": 45,
            "paddingLeft": 25,
            "paddingRight": 25,
            "width": "85%",
          },
          undefined,
        ]
      }
      testID="signup-password-input"
      value="pass"
    />
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 14,
            "lineHeight": 20,
          },
          {
            "color": "red",
            "fontSize": 11,
            "margin": -5,
          },
        ]
      }
      testID="signup-password-input-error"
    >
      min length is 6
    </Text>
    <View
      style={
        [
          {
            "backgroundColor": "#f6f6f6",
          },
          {
            "flexDirection": "row",
          },
        ]
      }
      testID="pet-view"
    >
      <View
        style={
          [
            {
              "backgroundColor": "#f6f6f6",
            },
            {
              "height": 45,
              "paddingRight": 2,
              "width": "30%",
            },
          ]
        }
        testID="pet-view"
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderBottomLeftRadius": 10,
              "borderColor": "#22D3EE",
              "borderTopLeftRadius": 10,
              "borderWidth": 1,
              "flex": 1,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID="pet-pick-country"
        >
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                {
                  "fontSize": 14,
                },
              ]
            }
            testID="pet-text"
          >
            ðŸ‡ºðŸ‡¸
             
            +1
          </Text>
        </View>
      </View>
      <TextInput
        autoCapitalize="none"
        defaultValue=""
        keyboardType="numeric"
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholderTextColor="#8e8e8e"
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "borderColor": "#22D3EE",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 45,
              "paddingLeft": 25,
              "paddingRight": 25,
              "width": "85%",
            },
            {
              "borderBottomLeftRadius": 0,
              "borderTopLeftRadius": 0,
              "width": "55%",
            },
          ]
        }
        testID="signup-phone-input"
        value="abc"
      />
    </View>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 14,
            "lineHeight": 20,
          },
          {
            "color": "red",
            "fontSize": 11,
            "margin": -5,
          },
        ]
      }
      testID="signup-phone-input-error"
    >
      cell phone must be numeric only
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#0e7490",
          "borderRadius": 10,
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
          "margin": 6,
          "opacity": 1,
          "padding": 8,
          "width": "85%",
        }
      }
      testID="pet-button"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#ffffff",
              "fontSize": 30,
            },
            [
              undefined,
            ],
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
        testID="pet-icon"
      >
        î³Ž
      </Text>
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            {
              "color": "#ffffff",
            },
          ]
        }
        testID="pet-text"
      />
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#f6f6f6",
          },
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 5,
          },
        ]
      }
      testID="pet-view"
    >
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            undefined,
          ]
        }
        testID="pet-text"
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="redirect-to-login-label"
      >
        <Text
          style={
            [
              {
                "color": "#155e75",
              },
              {
                "color": "#0a7ea4",
                "fontSize": 16,
                "lineHeight": 30,
              },
              {
                "textDecorationLine": "underline",
              },
            ]
          }
          testID="pet-title"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`<Signup /> Signup renders correctly and Snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#f6f6f6",
      },
      {
        "width": "100%",
      },
    ]
  }
  testID="pet-view"
>
  <View
    style={
      [
        {
          "backgroundColor": "#f6f6f6",
        },
        {
          "alignItems": "center",
          "gap": 5,
          "justifyContent": "center",
        },
      ]
    }
    testID="pet-view"
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          {
            "paddingHorizontal": 50,
            "paddingVertical": 20,
          },
        ]
      }
      testID="pet-text"
    >
      header_sign_up_form_message
    </Text>
    <TextInput
      autoCapitalize="none"
      defaultValue=""
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="header_sign_up_form_email"
      placeholderTextColor="#8e8e8e"
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "borderColor": "#22D3EE",
            "borderRadius": 10,
            "borderWidth": 1,
            "height": 45,
            "paddingLeft": 25,
            "paddingRight": 25,
            "width": "85%",
          },
          undefined,
        ]
      }
      testID="signup-email-input"
      value=""
    />
    <TextInput
      autoCapitalize="none"
      defaultValue=""
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="header_sign_up_form_password"
      placeholderTextColor="#8e8e8e"
      secureTextEntry={true}
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "borderColor": "#22D3EE",
            "borderRadius": 10,
            "borderWidth": 1,
            "height": 45,
            "paddingLeft": 25,
            "paddingRight": 25,
            "width": "85%",
          },
          undefined,
        ]
      }
      testID="signup-password-input"
      value=""
    />
    <View
      style={
        [
          {
            "backgroundColor": "#f6f6f6",
          },
          {
            "flexDirection": "row",
          },
        ]
      }
      testID="pet-view"
    >
      <View
        style={
          [
            {
              "backgroundColor": "#f6f6f6",
            },
            {
              "height": 45,
              "paddingRight": 2,
              "width": "30%",
            },
          ]
        }
        testID="pet-view"
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderBottomLeftRadius": 10,
              "borderColor": "#22D3EE",
              "borderTopLeftRadius": 10,
              "borderWidth": 1,
              "flex": 1,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID="pet-pick-country"
        >
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                {
                  "fontSize": 14,
                },
              ]
            }
            testID="pet-text"
          >
            ðŸ‡ºðŸ‡¸
             
            +1
          </Text>
        </View>
      </View>
      <TextInput
        autoCapitalize="none"
        defaultValue=""
        keyboardType="numeric"
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="header_sign_up_form_phone"
        placeholderTextColor="#8e8e8e"
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "borderColor": "#22D3EE",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 45,
              "paddingLeft": 25,
              "paddingRight": 25,
              "width": "85%",
            },
            {
              "borderBottomLeftRadius": 0,
              "borderTopLeftRadius": 0,
              "width": "55%",
            },
          ]
        }
        testID="signup-phone-input"
        value=""
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#0e7490",
          "borderRadius": 10,
          "flexDirection": "row",
          "gap": 8,
          "justifyContent": "center",
          "margin": 6,
          "opacity": 1,
          "padding": 8,
          "width": "85%",
        }
      }
      testID="pet-button"
    >
      <Text />
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            {
              "color": "#ffffff",
            },
          ]
        }
        testID="pet-text"
      >
        header_sign_up
      </Text>
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#f6f6f6",
          },
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 5,
          },
        ]
      }
      testID="pet-view"
    >
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            undefined,
          ]
        }
        testID="pet-text"
      >
        header_sign_up_account_message
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="redirect-to-login-label"
      >
        <Text
          style={
            [
              {
                "color": "#155e75",
              },
              {
                "color": "#0a7ea4",
                "fontSize": 16,
                "lineHeight": 30,
              },
              {
                "textDecorationLine": "underline",
              },
            ]
          }
          testID="pet-title"
        >
          header_log_in
        </Text>
      </View>
    </View>
  </View>
</View>
`;
